### DISCUSSION TITLE
using math function like tan, cos, sin

### DISCUSSION THREAD

**Discussion by henryding-emcore** [polarfire-soc/polarfire-soc-documentation#222] [Bare metal embedded software] [answered]
I’m using math function like tan, cos, sin in one of my .c files. I have included <math.h> in the file but still got error: 
undefined reference to `tan'
undefined reference to `cos'

I searched online and most of suggestions are to add -lm in the end of gcc command. Something like: gcc -pedantic -Wall -o theaded_program pth.c -pthread -lm
But I’m using Softconsole GUI.  How do I fix this problem?

**Comment by rockoguido** [comment, marked as answer]
looks like adding -lm to the libs via linker settings would sort this out in your SoftConsole based project.   
In particular Project -> C/C++ Build -> Settings -> Cross Linker -> Libraries -> Add and just add m

**Reply by henryding-emcore** [reply]
Thanks! The solution works!
